gradoop-data-integration/src/test/java/org/gradoop/dataintegration/transformation/impl/ExtractPropertyFromVertexTest.java

```diff
@@ -229,4 +230,44 @@ public void listPropertyTest() throws Exception {
     Assert.assertTrue(properties.containsAll(Arrays.asList("m", "n", "x", "y", "z")));
   }
 
+  /**
+   * Tests graph head ids get assigned to newly created elements.
+   *
+   * @throws Exception If collect doesn't work as expected.
+   */
+  @Test
+  public void containGraphHeadIdTest() throws Exception {
+    LogicalGraph socialGraph = getSocialNetworkLoader().getLogicalGraph();
+
+    // this operation should create 6 vertices with the property values:
+    // 'Berlin', 3 times 'Dresden', 2 times 'Leipzig'
+    ExtractPropertyFromVertex extract = new ExtractPropertyFromVertex("Person", "city",
+      "City", "name", EdgeDirection.ORIGIN_TO_NEWVERTEX, "livesIn");
+
+    LogicalGraph extractedGraph = socialGraph.callForGraph(extract);
+
+    List<EPGMVertex> createdVertices = new ArrayList<>();
+    extractedGraph
+      .getVerticesByLabel("City")
+      .output(new LocalCollectionOutputFormat<>(createdVertices));
+
+    List<EPGMEdge> createdEdges = new ArrayList<>();
+    extractedGraph
+      .getEdgesByLabel("livesIn")
+      .output(new LocalCollectionOutputFormat<>(createdEdges));
+
+    getConfig().getExecutionEnvironment().execute();
+
+    // get only graph id from graph head
+    GradoopId socialGraphId = socialGraph.getGraphHead().collect().get(0).getId();
```

Comment:
`collect()` calls `execute()` again, I think you can remove the call to `execute()`.
If that does not work you can use the `LocalCollectionOutputFormat` for the graph heads too.

Reply:
Good comment. I guess you can just use the `LocalCollectionOutputFormat` to receive the GraphHead. Does not break the code style and should be executed correctly.

Reply:
I adapted the test to obtain the graph head via LocalCollectionOutputFormat